TrafficSimulator/Assets/CarSpawner.cs

```diff
@@ -0,0 +1,254 @@
+                {
+                    // Calculate the length of the lane
+                    laneLength = laneLength + Vector3.Distance(_laneNodeTemp.Position, _laneNodeTemp.Next.Position);
+                    _laneNodeTemp = _laneNodeTemp.Next;
+                }
+                _lengths.Add(laneLength);
+            }
+        }
+
+        private void CalculateLaneRatios()
+        {
+            float totalLength = 0;
+
+            // Loop through all lane lengths
+            for (int i = 0; i < _lengths.Count; i++)
+            {
+                totalLength = totalLength + _lengths[i];
+            }
+
+            // Loop through all lane lengths
+            for (int i = 0; i < _lengths.Count; i++)
+            {
+                _ratios.Add(_lengths[i] / totalLength);
+            }
+        }
+
+        private void CalculateLaneIndexes()
+        {
+            int laneIndex;
+
+            // Loop through all roads
+            for (int i = 0; i < _roadSystem.RoadCount; i++)
+            {
+                _roads[i].OnChange();
+                laneIndex = 0;
+
+                // Loop through all lanes
+                for (int j = 0; j < _roads[i].LaneCount; j++)
+                {
+                    _indexes.Add(laneIndex);
+                    laneIndex = laneIndex + 1;
+                }
+            }
+        }
+
+        private void CalculateMaxCarsForLanes()
+        {
+            // Loop through all lanes
+            for (int j = 0; j < _lanes.Count; j++)
+            {
+                // Calculate the max cars for the lane
+                _maxCarsPerLane.Add((int)(_lengths[j] / CarLength));
+            }
+        }
+
+        private void SpawnCarsWithPercentage(float MaxCarsPercentage) 
+        {
+            for (int i = 0; i < _lanes.Count; i++)
+            {
+                // Calculate the number of cars to spawn
+                int carsToSpawn = (int)Mathf.Ceil(_maxCarsPerLane[i] * MaxCarsPercentage);
+                Debug.Log("Cars to spawn: " + carsToSpawn + " on lane " + i + " with " + _maxCarsPerLane[i] + " max cars");
+
+                // Calculate the _offset
+                _offset = _lanes[i].StartNode.Count / carsToSpawn;
+                _laneNodeCurrent = _lanes[i].StartNode;
+
+                // Spawn cars
+                for (int j = 0; j < carsToSpawn; j++)
+                {
+                    // Spawn individual car at current node
+                    SpawnCar(i);
+
+                    _carCounter = _carCounter + 1;
+
+                    Set_Offset(_offset);
+                }
+            }
+        }
+
+        private void SpawnCarsWithRatio(int maxCars)
+        {
+            int cars_Spawned = 0;
+            for (int i = 0; i < _lanes.Count; i++)
+            {
+                // Calculate the number of cars to spawn for this lane
+                int carsToSpawn = (int)Mathf.Ceil(_ratios[i] * maxCars);
```

Comment:
```suggestion
                int carsToSpawn = Mathf.CeilToInt(_ratios[i] * maxCars);
```
Try this instead so you don't have to cast